cmd/ci-operator/main.go

```diff
@@ -215,7 +212,8 @@ func bindOptions(flag *flag.FlagSet) *options {
 	// the target namespace and cleanup behavior
 	flag.StringVar(&opt.namespace, "namespace", "", "Namespace to create builds into, defaults to build_id from JOB_SPEC. If the string '{id}' is in this value it will be replaced with the build input hash.")
 	flag.StringVar(&opt.baseNamespace, "base-namespace", "stable", "Namespace to read builds from, defaults to stable.")
-	flag.DurationVar(&opt.idleCleanupDuration, "delete-when-idle", opt.idleCleanupDuration, "If no pod is running for longer than this interval, delete the namespace. Set to zero to retain the contents.")
+	flag.DurationVar(&opt.idleCleanupDuration, "delete-when-idle", opt.idleCleanupDuration, "If no pod is running for longer than this interval, delete the namespace. Set to zero to retain the contents. Requires the namespace TTL controller to be deployed.")
+	flag.DurationVar(&opt.cleanupDuration, "delete-after", opt.cleanupDuration, "If namespace exists for longer than this interval, delete the namespace. Set to zero to retain the contents. Requires the namespace TTL controller to be deployed.")
```

Comment:
What happens if the TTL controller is *not* deployed? This is annotation-driven, so I assume that the namespace will just live forever, right? Can we detect it and error out or at least warn?

Reply:
Correct, nothing would remove the namespace.

On the other hand, if it's not deployed, it's most likely a test cluster where
leaking namespaces is not a problem.

Reply:
The first case isn’t true.  It’s trivial easy to run without our infra, and we should keep it that way.  Doesn’t mean we need to support teardown when controller is not present, but don’t apply that philosophy to other parts.

Reply:
Eh, using `ci-operator` to run your unit tests is trivial. Getting all of the templates, configs and secrets together to run a templatized test on your own _is_ hard, and we are still working through the best way of fixing that. Even Gabe gave up.